xds/src/test/java/io/grpc/xds/XdsNameResolverProviderTest.java

```diff
@@ -82,4 +86,31 @@ public void newNameResolver() {
             args))
         .isNull();
   }
+
+  @Test
+  public void validName_withAuthority() {
+    XdsNameResolver resolver =
+        provider.newNameResolver(
+            URI.create("xds-experimental://trafficdirector.google.com/foo.googleapis.com"), args);
+    assertThat(resolver).isNotNull();
+    assertThat(resolver.getServiceAuthority()).isEqualTo("foo.googleapis.com");
+  }
+
+  @Test
+  public void validName_noAuthority() {
+    XdsNameResolver resolver =
+        provider.newNameResolver(URI.create("xds-experimental:///foo.googleapis.com"), args);
+    assertThat(resolver).isNotNull();
+    assertThat(resolver.getServiceAuthority()).isEqualTo("foo.googleapis.com");
+  }
+
+  @Test
+  public void invalidName_hostnameContainsUnderscore() {
+    try {
+      provider.newNameResolver(URI.create("xds-experimental:///foo_bar.googleapis.com"), args);
```

Comment:
nit: move `URI uri = URI.create("xds-experimental:///foo_bar.googleapis.com")` out of `try{}` block.

Reply:
Done.